cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -105,6 +105,9 @@ struct ARROW_EXPORT AzureOptions {
 
   Status ConfigureDefaultCredential(const std::string& account_name);
 
+  Status ConfigureManagedIdentityCredential(const std::string& account_name,
+                                            std::string const& clientId = std::string());
```

Comment:
Code Style: should be `client_id` and you can put the `const` before `std::string` like you did in the implementation. The MS C++ style is so weird :D

Reply:
I did it again :disappointed:. Fixed now.